kaldo/forceconstants.py

```diff
@@ -235,3 +211,45 @@ def unfold_third_order(self, reduced_third=None, distance_threshold=None):
         expanded_third = expanded_third.reshape(
             (n_unit_atoms * 3, n_replicas * n_unit_atoms * 3, n_replicas * n_unit_atoms * 3))
         return expanded_third
+
+
+    def elastic_prop(self):
+
+        # Intake key parameters
+        atoms = self.atoms
+        masses = atoms.get_masses()
+        volume = atoms.get_volume()
+        list_of_replicas = self.second.list_of_replicas
+        h0 = HarmonicWithQ(np.array([0, 0, 0]), self.second, storage='numpy')
+        dynmat = self.second.dynmat[0]  # units THz^2
+        positions = self.atoms.positions
+        n_unit = atoms.positions.shape[0]
+        e_mu = np.array(h0._eigensystem[1:, :]).reshape(
+            (n_unit, 3, 3 * (n_unit)))
+        w_mu = np.abs(np.array(h0._eigensystem[0, :])) ** (0.5)  # optical frequencies (w/(2*pi) = f) in THz
+        distance = positions[:, np.newaxis, np.newaxis, :] - (
+                    positions[np.newaxis, np.newaxis, :, :] + list_of_replicas[np.newaxis, :, np.newaxis, :])
+        d1 = np.einsum('iljx,ibljc->ibjcx',
+                       tf.convert_to_tensor(distance.astype(complex)),
+                       tf.cast(dynmat, tf.complex128))  # THz^2*Ang (it should be multiplied by i?)
```

Comment:
@ZKC19940412 why is it using `tf` here?

Reply:
@dafolkner might provide a better answer here, but I think since the second order force constants are loaded from the forceconstant object, so it might be handled better with tensorflow.

Reply:
in this case, it's probably better to use `opt_einsum.contract` and specify the engine to be `tensorflow`.